File: python-track/how-many-numbers-are-smaller-than-the-current-number.py
class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
        sorted_nums = sorted(nums)
        lookup = defaultdict(list)
        for i in range(len(nums)):
            lookup[sorted_nums[i]].append(i)

        ans = [0] * len(nums)
        for i in range(len(nums)):
            ans[i] = lookup[nums[i]][0]
        
        return ans
        

File: 1482-how-many-numbers-are-smaller-than-the-current-number/how-many-numbers-are-smaller-than-the-current-number.java
class Solution {
    public int[] smallerNumbersThanCurrent(int[] nums) {
        int[] sortedArr = Arrays.copyOf(nums, nums.length);
        Arrays.sort(sortedArr);
        System.out.println(sortedArr);
        int[] ans = new int[nums.length];
        for (int i = 0; i < nums.length; i++){
            for (int j = 0; j < nums.length; j++){
                if (sortedArr[j] == nums[i])
                    {
                        ans[i] = j;
                        break;
                    }
            }
        }
        
        return ans;
    }
}